File: typescript/tmps/tmp9/src/main.ts

function main(): void {
    const nums = [1, 2, 3, 4];
    const target = 7;
    const result = run(nums, target);

    console.log("Result: ", result);
}

function run(nums: number[], target: number): [number, number] | null {
    if (nums.length < 2) {
        return null
    }

    const seen = new Map<number, number>(); // complements; int to idx

    for (let i = 0; i < nums.length; i++) {
        const num = nums[i];
        const complement = target - num;

        if (seen.has(complement)) {
            const compIdx = seen.get(complement);
            return [compIdx, i];
        }
        seen.set(num, i);
    }

    return null
}

main()

File: python/tmps/tmp9/main.py

def main():
    nums: list = [1, 2, 3, 4]
    target: int = 7
    result: list = run(nums, target)

    print(f"Result: {result}")


def run(nums: list, target: int) -> list:
    seen: dict = {}

    for idx, num in enumerate(nums):
        complement = target - num
        if complement in seen:
            compIdx = seen[complement]
            return [compIdx, idx]
        seen[num] = idx

    return []


if __name__ == "__main__":
    main()
